subworkflows/local/bam_baserecalibrator/main.nf

```diff
@@ -9,33 +9,23 @@ include { GATK4_GATHERBQSRREPORTS } from '../../../modules/nf-core/gatk4/gatherb
 
 workflow BAM_BASERECALIBRATOR {
     take:
-        cram            // channel: [mandatory] meta, cram_markduplicates, crai
-        dict            // channel: [mandatory] dict
-        fasta           // channel: [mandatory] fasta
-        fasta_fai       // channel: [mandatory] fasta_fai
-        intervals       // channel: [mandatory] intervals, num_intervals
-        known_sites     // channel: [optional]  known_sites
-        known_sites_tbi // channel: [optional]  known_sites_tbi
+    cram            // channel: [mandatory] meta, cram_markduplicates, crai
+    dict            // channel: [mandatory] dict
+    fasta           // channel: [mandatory] fasta
+    fasta_fai       // channel: [mandatory] fasta_fai
+    intervals       // channel: [mandatory] intervals, num_intervals
+    known_sites     // channel: [optional]  known_sites
+    known_sites_tbi // channel: [optional]  known_sites_tbi
 
     main:
-    ch_versions = Channel.empty()
+    versions = Channel.empty()
 
     cram_intervals = cram.combine(intervals)
         .map{ meta, cram, crai, intervals, num_intervals ->
-
-            //If no interval file provided (0) then add empty list
-            intervals_new = num_intervals == 0 ? [] : intervals
-
-            [[
-                data_type:      meta.data_type,
-                id:             meta.sample,
-                num_intervals:  num_intervals,
-                patient:        meta.patient,
-                sample:         meta.sample,
-                sex:            meta.sex,
-                status:         meta.status,
-            ],
-            cram, crai, intervals_new]
+        //If no interval file provided (0) then add empty list
+        [ meta.subMap('data_type', 'patient', 'sample', 'sex', 'status')
```

Comment:
love the submap, but It would be great if we would find an option in the future to just touch the fields that need changing. Still now everytime we touch the meta fields we have to go through all the code again.

Reply:
If you're not going to be removing any entries from the map, only modifying them, you might consider doing something like

```groovy
meta + [ id:meta.sample, num_intervals:num_intervals ]
```

The precedence is such that the map on the right-hand side takes precedence, so if there are any keys common between the two maps being added, the value in the right-hand map is the one used in the final returned map.

Would that address your concerns?

Reply:
If you also need to remove entries, the same idea applies:

```groovy
meta + [ id:meta.sample, num_intervals:num_intervals ] - meta.subMap('anotherKey')
```